tests/unit/cunumeric/test_config.py

```diff
@@ -0,0 +1,236 @@
+        assert lib.name == "foo"
+        assert lib.shared_object is None
+        assert lib.runtime is None
+
+    def test_get_name(self) -> None:
+        lib = m.CuNumericLib("foo")
+        assert lib.get_name() == "foo"
+
+    def test_get_shared_library(self) -> None:
+        lib = m.CuNumericLib("foo")
+        result = lib.get_shared_library()
+        assert isinstance(result, str)
+
+        from cunumeric.install_info import libpath
+
+        assert result.startswith(libpath)
+
+        assert "libcunumeric" in result
+
+        assert result.endswith(lib.get_library_extension())
+
+    def test_get_c_header(self) -> None:
+        lib = m.CuNumericLib("foo")
+
+        from cunumeric.install_info import header
+
+        assert lib.get_c_header() == header
+
+    def test_get_registration_callback(self) -> None:
+        lib = m.CuNumericLib("foo")
+        assert (
+            lib.get_registration_callback() == "cunumeric_perform_registration"
+        )
+
+    def test_initialize(self) -> None:
+        lib = m.CuNumericLib("foo")
+        lib.initialize(_FakeSO)
+        assert lib.shared_object == _FakeSO
+
+        # error if runtime already set
+        lib.runtime = runtime
+        with pytest.raises(AssertionError):
+            lib.initialize(_FakeSO)
+
+    def test_set_runtine(self) -> None:
+        lib = m.CuNumericLib("foo")
+
+        # error if not initialized
+        with pytest.raises(AssertionError):
+            lib.set_runtime(runtime)
+
+        lib.initialize(_FakeSO)
+        lib.set_runtime(runtime)
+        assert lib.runtime == runtime
+
+        # error if runtime already set
+        with pytest.raises(AssertionError):
+            lib.set_runtime(runtime)
+
+    def test_get_resource_configuration(self) -> None:
+        lib = m.CuNumericLib("foo")
+
+        # error if not initialized
+        with pytest.raises(AssertionError):
+            lib.get_resource_configuration()
+
+        lib.initialize(_FakeSO)
+        config = lib.get_resource_configuration()
+        assert isinstance(config, ResourceConfig)
+        assert config.max_tasks == _FakeSO.CUNUMERIC_MAX_TASKS
+        assert config.max_mappers == _FakeSO.CUNUMERIC_MAX_MAPPERS
+        assert config.max_reduction_ops == _FakeSO.CUNUMERIC_MAX_REDOPS
+        assert config.max_projections == 0
+        assert config.max_shardings == 0
+
+    @patch("cunumeric.runtime.destroy")
+    def test_destroy(self, mock_destroy) -> None:
+        lib = m.CuNumericLib("foo")
+        lib.initialize(_FakeSO)
+        lib.set_runtime(runtime)
+        lib.destroy()
+        assert mock_destroy.called_once_with()
+
+
+def test_CUNUMERIC_LIB_NAME() -> None:
+    assert m.CUNUMERIC_LIB_NAME == "cunumeric"
+
+
+def test_cunumeric_lib() -> None:
+    assert isinstance(m.cunumeric_lib, m.CuNumericLib)
+
+
+def test_cunumeric_context() -> None:
+    assert isinstance(m.cunumeric_context, Context)
+
+
+def test_CuNumericOpCode() -> None:
```

Comment:
Excuse my naiveté, but I don't see what value such a testcase is adding. It's simply re-listing all the members of the original enum.

Reply:
It's just a roadblock. It protects against someone unintentionally changing the enum, or intentionally changing it without understanding or considering consequences. Ideally that sort of thing is flagged in review, but sometimes PRs are big and people are busy, and making it so the test has to be updated makes it harder to miss.